### DISCUSSION TITLE
VPN Mode / Exit Node seems to be broken

### DISCUSSION THREAD

**Discussion by rjc3rd** [oxen-io/lokinet#2208] [Q&A]
I used to use Lokinet VPN only occasionally, a little over a year ago, and it always worked when I opened it and started the VPN mode on exit.loki exit node.

Recently I reinstalled it and was a little surprised to find that when I enabled VPN mode, I lost all internet access. I can't ping 1.1.1.1 or open any website with VPN mode enabled.

I've only ever used it for the VPN mode, by the way, and I'm running Linux Mint Cinnamon 21.2 if that matters. 

To test and attempt to fix it, I reinstalled it a few times but it just wouldn't work. After that I installed to a Linux Mint MATE 21.2 QEMU/KVM with the same result. Fresh install of LM and Lokinet all default install. No internet access with VPN enabled. I thought it might be related to Linux Mint since it requires a small workaround in the oxen.list file to get it to install, so I installed Ubuntu 23.04 also in a KVM, and it still would not access internet with VPN enabled. Also new clean default install and this time with no work arounds.

So, I'm not saying that I think the software is broken. I find it hard to believe that the latest version has been up for almost a year and this feature is not working. I believe if the software was broken, it would have been fixed by now. This is the reason that I posted in Discussions instead of Issues.

My thought is that maybe the exit node exit.loki may be flooded with users and not able to access clearnet. 

Am I the only one? Any suggestions? Thanks.

**Comment by rjc3rd** [comment]
Also, I have two ISPs at the house, of course on two different routers. I tried both ISPs during my testing. Internet was always working during the tests. Internet only failed when VPN move was enabled.

When I tested the KVMs, I made sure the host had good internet first. then verified that the KVM had good internet access before enabling VPN mode. While the KVM Lokinet VPN was not working, the host still had access to internet.

**Comment by rjc3rd** [comment]
https://www.privacyproshop.com/product/lokinet-exit-node-access-code-60-day/
Reluctantly, I paid the fee. I purchased a 30 day Exit Node.

Here are my results.

lokinet-vpn --down

ping google.com
PING google.com (142.250.114.102) 56(84) bytes of data.
64 bytes from rr-in-f102.1e100.net (142.250.114.102): icmp_seq=1 ttl=56 time=20.3 ms
64 bytes from rr-in-f102.1e100.net (142.250.114.102): icmp_seq=2 ttl=56 time=20.9 ms
64 bytes from rr-in-f102.1e100.net (142.250.114.102): icmp_seq=3 ttl=56 time=22.3 ms
--- google.com ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 20.276/21.147/22.299/0.849 ms

ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
64 bytes from 1.1.1.1: icmp_seq=1 ttl=56 time=15.4 ms
64 bytes from 1.1.1.1: icmp_seq=2 ttl=56 time=15.7 ms
64 bytes from 1.1.1.1: icmp_seq=3 ttl=56 time=15.1 ms
64 bytes from 1.1.1.1: icmp_seq=4 ttl=56 time=12.9 ms
--- 1.1.1.1 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3006ms
rtt min/avg/max/mdev = 12.924/14.784/15.727/1.096 ms

lokinet-vpn --up --exit exitnodeprovidedbyprivacyproshop.loki --auth passwordprovidedbyprivacyproshop

ping google.com
ping: google.com: Name or service not known

ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
--- 1.1.1.1 ping statistics ---
19 packets transmitted, 0 received, 100% packet loss, time 18428ms

Privacy Pro Shop email says to wait upto 15 for access code to become valid.
I have now waited over half an hour.

I'll try again later to see if there is just more lag time than suggested.

I'll update if anything changes.

**Comment by rjc3rd** [comment]
UPDATE

I rebooted and tried again.

It seems to be working now. Not sure if it was related to the time needed to wait longer or just restarting the daemon.

Although after purchasing the exit node I did --down and --up the vpn, but I did not stop and restart the daemon.

Rebooting certainly did restart the daemon and now all is working correctly.

I have 30 days to test this out. I will continue to update this thread, since I suspect others could use this info.

**Comment by rjc3rd** [comment]
https://github.com/oxen-io/lokinet/issues/2185#issuecomment-1567482032

I found where @majestrate mentioned that the exit node exit.loki is/was down.

Possibly later it will come back up soon. I really don't use the VPN mode enough to pay $8 a month for it. At lease now I've seen a comment by a dev saying that exit.loki is not (or at lease was not) working.

**Comment by rjc3rd** [comment]
https://oxen.directory/exitnodes/

I found this list of exit nodes if you want to try.
I did NOT try them when I was troubleshooting.

Known free Lokinet Exit Nodes (Verified on April 3, 2023)
    exit.loki     – located in the US
    exit2.loki    – located in the US
    euroexit.loki – located in the Netherlands
    door.loki     – located in the US

**Comment by rjc3rd** [comment]
Initially, after purchasing an exit node from Privacy Pro Shop, VPN mode seemed to be working. I left it on for maybe an hour or so. I was even able to stream video from bitchute during that time. Then it just quit working again. I tried --down and --up the vpn a few times but it never started working again. I've even uninstalled and reinstalled a couple of times.

One thing of note, since the beginning for me, I have been able to confirm that at all times, .loki sites are working. when I --down the vpn, I get regular ol' internet access and .loki sites in the same browser. When I --up the vpn, I am still able to browse .loki sites but no clearnet sites. 

ping 1.1.1.1 NOT working
ping google.com NOT working
ping nemo.loki WORKING fine.

I'm just over here on github talking to myself about this though, so not sure if any of the devs even care about vpn or exit node problems. I'm pretty sure that's not the main purpose of the network anyway, so I don't know.

So, anyway, I'll just leave this here.

**Comment by rjc3rd** [comment]
I sent an email to Privacy Pro Shop this evening to ask for suggestions.

I sent my email pretty late, just after 5pm and got a reply in only a couple of hours.

He informed me that he has been using a node in the Netherlands and connected to the Montreal node as a test for me. I think he said that he even tested Phoenix too.

His connections were working all afternoon. I tried again and now I am able to use VPN mode again. I didn't make any changes to my settings today. It's just working now. Also, in every case, I was able to log into the node, I just didn't have clearnet access, but I do now. He also confirmed that exit.loki is still not working. 

I will continue to test this out.

Thanks Tim, I appreciate the quick response.

**Comment by KeeJef** [comment]
Lokinet is quite fragile in its current state, we are making a number of improvements internally which should make Lokinet more reliable, not sure about this specific issue, some free exits are no longer available, but beyond that it sounds like you might be getting some bad paths. 

We have been laser focused on making internal improvements to Lokinet to improve speeds and reliability, once these internal overhauls are made we intend to circle back on consumer facing and application facing developments.

**Comment by rjc3rd** [comment]
@KeeJef Thanks for the update.

I have continued to test this out over here. It does seem to be working for the most part. It does occasionally lose connectivity to clearnet, and even a few times to lokinet completely, but I am able to get right back on. I can even stream video while connected via Lokinet VPN. 

I watched a video today from Tim Sutinen of Privacy Pro Shop talking about some changes going on over at Session/Lokinet/Oxen that I think might possibly be having something to do with the instability I am experiencing lately along with the comment from @KeeJef about the current state of Lokinet. 

The video if you are interested.
https://odysee.com/@PrivacyProShop:4/OXEN-to-SENT-transition-news:2

I have tested the 4 known free exit nodes as well
exit.loki will connect but there is no data transfer at all to clearnet.
exit2.loki error message "could not add exit"
door.loki error message "we could not find an exit with that name"
euroexit.loki working but very slow - speed test results 0.15 Mbps

I've decided to hang in there and wait for it to get better. I want lokinet to work.
